src/Uuid/UuidGenerator.php

```diff
@@ -33,12 +33,12 @@ public function generateV4() {
      *
      * @return String   Valid v5 UUID.
      */
-    public function generateV5($str, $namespace = NULL) {
+    public function generateV5($namespace = NULL) {
         // Use default namespace if none is provided.
         if (empty($namespace)) {
             $namespace = $this->namespace;
         }
 
-        return Uuid::uuid5($namespace, $str)->toString();
+        return Uuid::uuid5(Uuid::NAMESPACE_DNS, $namespace)->toString();
```

Comment:
Dani, just to be sure, without the string you get always the same if using Uuid::NAMESPACE_DNS + domain
So this is a salt. and makes unique base UUID for your own domain.  Uuid::NAMESPACE_DNS = is a constant UUID defined in RFC.
If you pass this output to the generator as first input and then the string (lets say an dc identifier) you get the final one. If not you will get always the same. So both of you are right, you need to run both. (but the first one should be static because if the config is not changing neither that one, the reason i put it so on the old silex version)
https://www.ietf.org/rfc/rfc4122.txt (page 18)